sandbox/libs/authn/src/main/java/org/opensearch/authn/internal/InternalSubject.java

```diff
@@ -22,8 +25,12 @@
 public class InternalSubject implements Subject {
     private final org.apache.shiro.subject.Subject shiroSubject;
 
+    public HashMap<String, ArrayList<Permission>> grantedPermissions; // Not sure how we plan to store the permissions
```

Comment:
To encapsulate the permissions, can we change this into an instance of the PermissionsHandler class?  Maybe permissions storage/store is a more appropriate name

Reply:
Currently the PermissionHandler is a static class so there is only one instance of it operating on numerous InternalSubjects and providing control of the different permissions each subject may have. This can be switched so that the PermissionHandler is not static and then this attribute could be defined as an instance if that is what you prefer.